### ISSUE TITLE
ASRFromScratch using locally trained LM

### ISSUE THREAD

**Issue by catskillsresearch** [speechbrain/speechbrain#1286]
The [template YAML for ASRFromScratch](https://github.com/speechbrain/speechbrain/blob/develop/templates/speech_recognition/ASR/train.yaml)  uses the pretrained LM model on line 38.   Following the [ASRFromScratch CoLab](https://colab.research.google.com/drive/1aFgzrUv3udM_gNJNUoLaHIm78QHtxdIz?usp=sharing), we make an LM model locally.  To use the local Tokenizer, the instructions in the YAML are to replace line 38 with "a local path pointing to a directory containing the lm.ckpt and tokenizer.ckpt".  However:

* The Tokenizer in Tokenizer/save contains 1000_unigram.model and 1000_unigram.vocab.  There is no file tokenizer.ckpt.  I found this clue, which I read as saying that I can ignore the 1000_unigram.vocab and rename the 1000_unigram.model to tokenizer.ckpt: https://giters.com/speechbrain/speechbrain/issues/1061.  I can code the tokenizer path to this at the bottom of the YAML, so OK there. ✔
* The LM has LM/results/RNNLM/save/CKPT+2022-01-31+10-42-23+01/model.ckpt.  There is no lm.ckpt.  It seems I can code the LM path at the bottom of the YAML to this model.ckpt file and that works.  So OK there.✔
* The LM config file RNNLM.yaml has embedding dimension 256, but ASR/train.yaml has cnn_channels: (128,256) on line 81 and emb_size: 128 on line 91.  It seems that one or both of these needs to be changed to 256 to be compatible with the locally trained LM.  Changing both to 256 seems to work. ✔
* The ASR config file references pretrained asr.ckpt.  Commenting that out causes the training process to fail.  So the demo CoLab is showing fine-tuning of a pre-existing model.  How do we create asr.ckpt from scratch?
* This started with me puzzling over how to do inference with the trained model.  I have many clues to follow on how to do that, it seems to be extremely hard.  I will update this note if I can get it together from the clues.

**Comment by TParcollet** [comment]
For point 1. Yes, it a mistake in the Colab .ckpt = .model; 2. Same; 3. Of course embeddings dim must be the same in the yaml :-) 4. Simply commenting "model: !ref <model>" in the PreTrainer in the YAML should do (haven't check in details, but theoretically it should be enough). The PreTrainer is where you will find things that are loaded as PRETRAINED models, you can just remove them from there if you don't want them to be loaded as pretrained. 5. It's extremely simple, if you simply follow the tutorial on Inference, you'll see that it's very simple. You can do it in 3 different ways ranging from development to productions settings.
